server/athenian/api/internal/features/metric_calculator.py

```diff
@@ -647,6 +648,63 @@ def reset(self) -> None:
         [c.reset() for c in self._calcs]
 
 
+class ThresholdComparisonRatioCalculator(AggregationMetricCalculator[float]):
+    """Calculate the ratio of metric values satisfying a given threshold.
+
+    Metric values are computed by the upstream MetricCalculator dependency.
+
+    Subclasses must specify the numpy comparison function
+    (eg. np.less_equal, np.greater, ...) to use between the metric
+    values and the threshold.
+
+    """
+
+    metric = make_metric("MetricComparisonRatio", __name__, np.float32, -1)
+
+    def compare(
+        self,
+        peek: np.ndarray,
+        threshold: Any,
+        *,
+        out: np.ndarray,
+        where: np.ndarray,
+    ) -> Callable:
```

Comment:
```suggestion
    ) -> np.ndarray:
```
right? We return an array - the result of the comparison.

Reply:
right fixed